pkg/apis/core/v1alpha1/types.go

```diff
@@ -129,6 +129,8 @@ type DataVolumeStatus struct {
 	Phase        DataVolumePhase    `json:"phase,omitempty"`
 	Progress     DataVolumeProgress `json:"progress,omitempty"`
 	RestartCount int32              `json:"restartCount"`
+	// +listType=set
+	Conditions []*DataVolumeCondition `json:"conditions,omitempty" optional:"true"`
```

Comment:
Unusual to have slice of pointers in k8s api.  `Conditions []DataVolumeCondition` preferred

Reply:
Yeah, was having issue with getting to the backing data of the slice, for some reason this:
```go
for _, cond := range conditions {
  return &cond
}
```
doesn't give me the backing data, but
```go
for i, cond := range conditions {
  return &conditions[i]
}
```
does

Reply:
In your first example, `cond` is a copy of the slice item

Reply:
Yeah I had initially not realized this, that is why I did the pointer, that way the pointer was the copy and I was able to modify the actual struct. Once I realized its a copy, and I could return the pointer in the second way there was no reason for the pointer slice.